packages/donor/src/screens/bookDonation/BookDonationScreenContainer.tsx

```diff
@@ -34,22 +42,59 @@ export function BookDonationScreenContainer({
   }
 
   const onSlotSelected = (donationSlot: DonationSlotToBook) => {
-    appointmentToBookStore.setAppointmentToBook(donationSlot);
-    if (isLoggedIn) {
-      navigate(MainNavigationKeys.Questionnaire);
+    setSelectedSlot(donationSlot);
+    const currentDonor = appState.donor;
+    const timeDifference = currentDonor?.lastCompletedDonationTimeMillis
+      ? DateUtils.getNumberOfDaysBetweenDates(
+          new Date(donationSlot.donationStartTimeMillis),
```

Comment:
Seems like you have the time in millis, why convert it to Date and then to millis again?

Reply:
All of the functions in DateUtils are accepting Date as parameters, I wanted the new function to align

Reply:
I see.

Notice some utils accept `Date | number`. If you change your util to do so it would accept both, and then you'd be able to call it here with the millis directly